XiaoMi/pegasus
# feat: overload dump_write_request of replication_app_base.h

### What problem does this PR solve? 
https://github.com/XiaoMi/rdsn/pull/414 forbids the large write size request, but the log information is simple, https://github.com/XiaoMi/rdsn/pull/458 provide the interface of `dump_write_request`, this pr overload it.

### Check List 

Tests 
- Manual test (add detailed scripts or steps below)
  -  start onebox with config `max_allowed_write_size = 5`
  -  use shell `set` any key-value whose size > 5
  -  check the log:
```
[2.4@127.0.0.1:34801] client from 127.0.0.1:1 write request body size exceed threshold, request=[put:hash_key=a,sort_key=abc], request_body_size = 99, max_allowed_write_size = 5, it will be rejected!
``` 

Related changes

- Need to cherry-pick to the release branch(**depends on https://github.com/XiaoMi/rdsn/pull/458 merged**)
- Need to be included in the release note
